eftcamb/03_abstract_EFTCAMB_cache.f90

```diff
@@ -409,6 +510,46 @@ subroutine EFTCAMBParameterCachePrint( self )
 
     end subroutine EFTCAMBParameterCachePrint
 
+   ! ---------------------------------------------------------------------------------------------
+   !> Subroutine that check if an element of the EFTCAMB_parameter_cache is Nan.
+   subroutine EFTCAMBParameterCacheIsNan( self, HaveNan )
+
+     implicit none
+
+     class(EFTCAMB_parameter_cache), intent(in)  :: self    !< The base class.
+     logical, intent(out)                        :: HaveNan !< Logical variable which describes the presence of a Nan variable.
+                                                            !< If an element of the EFTCAMB_parameter_cache is Nan, you get HaveNan=.True.
+     logical                                     :: NanMass
+     integer                                     :: i
+
+     HaveNan = .False.
+     HaveNan = HaveNan.or.IsNaN(self%omegac)
+     HaveNan = HaveNan.or.IsNaN(self%omegab)
+     HaveNan = HaveNan.or.IsNaN(self%omegav)
+     HaveNan = HaveNan.or.IsNaN(self%omegak)
+     HaveNan = HaveNan.or.IsNaN(self%omegan)
+     HaveNan = HaveNan.or.IsNaN(self%omegag)
+     HaveNan = HaveNan.or.IsNaN(self%omegar)
+     HaveNan = HaveNan.or.IsNaN(self%h0)
+     HaveNan = HaveNan.or.IsNaN(self%h0_Mpc)
+     HaveNan = HaveNan.or.IsNaN(self%grhog)
+     HaveNan = HaveNan.or.IsNaN(self%grhornomass)
+     HaveNan = HaveNan.or.IsNaN(self%grhoc)
+     HaveNan = HaveNan.or.IsNaN(self%grhob)
+     HaveNan = HaveNan.or.IsNaN(self%grhov)
+     HaveNan = HaveNan.or.IsNaN(self%grhok)
+     HaveNan = HaveNan.or.IsNaN(self%Num_Nu_Massive*1.0_dl)
+     HaveNan = HaveNan.or.IsNaN(self%Nu_mass_eigenstates*1.0)
+     !
+     NanMass= .False.
```

Comment:
No need to use a new variable. It's cleaner fi we continue using HaveNan.
After this small mod the file is OK.